### ISSUE TITLE
Bug in TransactionPool.transactionInPool

### ISSUE THREAD

**Issue by vekexasia** [LiskHQ/lisk#707]
Hello,

The current way `transactionInPool` checks if the provided tx is in pool or not contains a minor bug with a very low severity.

This is the current code:

```js
 TransactionPool.prototype.transactionInPool = function (id) {
	return [
		self.unconfirmed.index[id],
		self.bundled.index[id],
		self.queued.index[id],
		self.multisignature.index[id]
	].filter(Boolean).length > 0;
};
```

Notice that the first queued transaction upon startup will be added at index `0` which will evaluate to `false` in the `.filter()` condition.

A safer way would be to check if array contains a number. In that case we know that the tx is actually in the pool.

Here below a screenshot showing what happens in case tx has 0 index with current and proposed implementation:

![image](https://user-images.githubusercontent.com/200523/29112135-3394aff8-7ced-11e7-833b-0d063b8df091.png)

**Comment by karmacoma** [comment]
As #725 was cancelled, we will look into this once more.

**Comment by 4miners** [comment]
Issues is already solved in 1.0.0 branch, however tests for it were overwritten (here https://github.com/LiskHQ/lisk/commit/6a93bfcbe3e3bd5f8fb241ac38053f3788efbf14). I will backport that change from 1.0.0 and add some tests.
